### PR TITLE
PHP 8.1 compatibility changes

### PR DESCRIPTION
**Pull request by jastraat**
Related issue: https://github.com/GetDKAN/dkan/issues/3898

This updates weitzman/drupal-test-traits. the drupal core recommended package still requires guzzle version 6 as do a handful of other 3rd party packages listed in the issue above, so this PR does not provide full compatibility with PHP 8.1, but once the core recommended package is updated (D10?) along with the other packages, DKAN should be good to go with PHP 8.1.

As mentioned in the above issue, I was able to get all automated tests to run with PHP 8.1 by manually updating the 3rd party libraries in my local installation. I also ran all automated tests with PHP 7.4 to make sure that these changes did not break sites using an older version of PHP.

Note that the auto_detect_line_endings is no longer supported in PHP 8.1, so this change removes that.

In addition, the Drupal core recommended package still requires guzzlehttp version 6.

Related packages that also require updates:

- https://github.com/fmizzell/maquina - Maquina\StateMachine\Machine::jsonSerialize()
- https://github.com/m1x0n/opis-error-presenter - PresentedValidationError::jsonSerialize()
- https://github.com/GetDKAN/csv-parser - CsvParser\Parser\Csv::jsonSerialize()
- https://github.com/GetDKAN/procrastinator - Result::jsonSerialize() and Job::jsonSerialize()

- [ ] Test coverage exists
- [ ] Documentation exists

### CODE REVIEW

**Comment by jastraat** [comment]
Drupal itself is not using the code sniffer checks disabled by this PR. They are excluded here: 

- https://git.drupalcode.org/project/drupal/-/blob/10.1.x/core/phpcs.xml.dist#L69
- https://git.drupalcode.org/project/drupal/-/blob/10.1.x/core/phpcs.xml.dist#L79

File: modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php [hunk 1]
```diff
@@ -89,15 +89,9 @@ protected function runIt() {
    */
   protected function getColsFromFile(string $file_path, string $delimiter): array {
 
-    // Ensure the "auto_detect_line_endings" ini setting is enabled before
-    // openning the file to ensure Mac style EOL characters are detected.
-    $old_ini = ini_set('auto_detect_line_endings', '1');
```
Comment (paul-m) [inline modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php:94]: Maybe make a note in the issue somewhere that this changes compatibility with \r line endings. https://php.watch/versions/8.1/auto_detect_line_endings-ini-deprecated

**Review by paul-m — CHANGES_REQUESTED** [review]
Generally LGTM but let's be sure and have a release note that says we don't support files with \r line endings any more.

File: modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php (hunk 1 above)
Comment (jastraat) [reply modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php:94]: Thanks @paul-m ! I don't see a change log file. Where would I update the release notes as part of this PR?

Comment (jastraat) [reply modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php:94]: Janette added a mention in the draft release notes for the next DKAN release here: https://github.com/GetDKAN/dkan/releases

**Review by janette — APPROVED** [review]
Adding won't fix exceptions to skip the warnings on #[\ReturnTypeWillChange] comments required by PHP 8.1

File: modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php (hunk 1 above)
Comment (janette) [reply modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php:94]: Will document in the release notes.

Comment (paul-m) [reply modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php:94]: Confirmed @janette drafted a release change. woot!

**Review by paul-m — APPROVED** [review]

**Comment by maietta** [comment]
Status?

**Pull request merged**